Mail/AppDelegate.swift

```diff
@@ -17,19 +17,31 @@
  */
 
 import CocoaLumberjackSwift
+import InfomaniakCore
 import InfomaniakDI
 import InfomaniakNotifications
 import MailCore
 import UIKit
 
-class AppDelegate: UIResponder, UIApplicationDelegate {
+@available(iOSApplicationExtension, unavailable)
+final class AppDelegate: UIResponder, UIApplicationDelegate {
     private let notificationCenterDelegate = NotificationCenterDelegate()
-    static var orientationLock = UIInterfaceOrientationMask.all
 
-    func application(
-        _ application: UIApplication,
-        willFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil
-    ) -> Bool {
+    @LazyInjectService private var orientationManager: OrientationManageable
+    @LazyInjectService private var accountManager: AccountManager
+    @LazyInjectService private var applicationState: ApplicationStatable
+
+    /// Making sure the DI is registered at a very early stage of the app launch.
+    private let dependencyInjectionHook = EarlyDIHook()
+
+    func application(_ application: UIApplication,
+                     willFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool
+    {
+        Logging.initLogging()
```

Comment:
Maybe we can move this line in AppAssembly to be sure that Logging is init as early as possible.

Reply:
I did not add it to `ApplicationAssembly,` but I added it in the init of `EarlyDIHook` before the DI is called.